mainhandler/imageregistryhandler.go

```diff
@@ -414,23 +427,42 @@ func (registryScan *registryScan) filterScanLimit(limit int) {
 	registryScan.mapImageToTags = filteredImages
 }
 
+func (registryScan *registryScan) enumerateRepoes() ([]string, error) {
+	// token sometimes includes a lot of dots, so we need to remove them
+	i := strings.Index(registryScan.registryInfo.AuthMethod.Password, ".....")
+	if i != -1 {
+		registryScan.registryInfo.AuthMethod.Password = registryScan.registryInfo.AuthMethod.Password[:i]
+	}
+
+	repos, err := registryScan.listReposInRegistry()
+	if err != nil {
+		return []string{}, err
+	}
+	return repos, nil
+
+}
+
 func (registryScan *registryScan) listReposInRegistry() ([]string, error) {
 	iRegistry, err := registryScan.makeRegistryInterface()
 	if err != nil {
 		return nil, err
 	}
 	regCreds := registryScan.registryCredentials()
-	var repos []string
+	var repos, pageRepos []string
+	var nextPage *regCommon.PaginationOption
 
 	firstPage := regCommon.MakePagination(iRegistry.GetMaxPageSize())
 	ctx := context.Background()
 	catalogOpts := regCommon.CatalogOption{IsPublic: !registryScan.hasAuth(), Namespaces: registryScan.registry.projectID}
-	for pageRepos, nextPage, err := iRegistry.Catalog(ctx, firstPage, catalogOpts, regCreds); ; pageRepos, nextPage, err = iRegistry.Catalog(ctx, *nextPage, catalogOpts, regCreds) {
+	for pageRepos, nextPage, err = iRegistry.Catalog(ctx, firstPage, catalogOpts, regCreds); ; pageRepos, nextPage, err = iRegistry.Catalog(ctx, *nextPage, catalogOpts, regCreds) {
 		if err != nil {
 			return nil, err
```

Comment:
Should we set the job report details to descriptive in what stage we had an error?
Like what is the content of the nextPage or something?

Reply:
If we get an error, we won't have nextPage. We do set details based on different errors for test connectivity

Reply:
Details based on steps/stages not on errors